src/service/resources-impl.js

```diff
@@ -1741,28 +1816,57 @@ export class Resource {
   }
 
   /**
-   * Calls element's documentInactiveCallback callback and resets state for
+   * Calls element's unlayoutCallback callback and resets state for
    * relayout in case document becomes active again.
    */
-  documentBecameInactive() {
-    if (this.state_ == ResourceState_.NOT_BUILT) {
+  unlayout() {
+    if (this.state_ != ResourceState_.LAYOUT_COMPLETE) {
       return;
     }
     if (this.isInViewport()) {
       this.setInViewport(false);
     }
-    if (this.element.documentInactiveCallback()) {
+    if (this.element.unlayoutCallback()) {
       this.state_ = ResourceState_.NOT_LAID_OUT;
       this.layoutCount_ = 0;
     }
   }
 
   /**
-   * Called when a previously visible element has become invisible.
+   * Calls element's pauseCallback callback.
+   */
+  pause() {
+    if (this.state_ == ResourceState_.NOT_BUILT) {
+      return;
+    }
+    if (this.isInViewport()) {
+      this.setInViewport(false);
+    }
+    this.element.pauseCallback();
+  }
+
+  /**
+   * TODO(jridgewell): This should eventually call layoutCallback
+   */
+  layout() {
+  }
+
+  /**
+   * Calls element's resumeCallback callback.
+   */
+  resume() {
+    if (this.state_ == ResourceState_.NOT_BUILT) {
+      return;
+    }
+    this.element.resumeCallback();
+  }
+
+  /**
+   * Called when a previously visible element is no longer displayed.
    */
   unload() {
-    // TODO(dvoytenko): Likely warrants its own callback and re-layout rules.
-    this.documentBecameInactive();
+    this.pause();
```

Comment:
Re: https://github.com/ampproject/amphtml/pull/1637#discussion_r55554213, https://github.com/ampproject/amphtml/pull/1637#discussion_r55552925. `#unload` should definitely be calling both `#pause` and `#unlayout`.

We can make `pauseAndUnlayout` just call `#unload` instead.